2002-12-20  Anton V. Denisov <avd@altlinux.org>

	* cmdline/apt-cdrom.cc (DoAdd):
	Simple workaround for ParseQuoteWord(Buffer,URI) == false
	if URI contains symbol '#'.
	ParseQuoteWord() need to be tested.

--- apt-0.5.4cnc9.orig/cmdline/apt-cdrom.cc	2002-12-14 14:35:00 +1200
+++ apt-0.5.4cnc9/cmdline/apt-cdrom.cc	2002-12-20 00:17:51 +1200
@@ -631,7 +631,7 @@ bool DoAdd(CommandLine &)
 	    // Escape special characters
 	    string::iterator J = Name.begin();
 	    for (; J != Name.end(); J++)
-	       if (*J == '"' || *J == ']' || *J == '[')
+	       if (*J == '"' || *J == ']' || *J == '[' || *J == '#')
 		  *J = '_';
 	    
 	    cout << "Found label '" << Name << "'" << endl;
@@ -650,7 +650,8 @@ bool DoAdd(CommandLine &)
 	    if (Name.empty() == false &&
 		Name.find('"') == string::npos &&
 		Name.find('[') == string::npos &&
-		Name.find(']') == string::npos)
+		Name.find(']') == string::npos &&
+		Name.find('#') == string::npos)
 	       break;
 	    cout << "That is not a valid name, try again " << endl;
 	 }	 
@@ -662,7 +663,7 @@ bool DoAdd(CommandLine &)
    // Escape special characters
    string::iterator J = Name.begin();
    for (; J != Name.end(); J++)
-      if (*J == '"' || *J == ']' || *J == '[')
+      if (*J == '"' || *J == ']' || *J == '[' || *J == '#')
 	 *J = '_';
    
    Database.Set("CD::" + ID,Name);
